ProteoformSuiteInternal/Proteoform.cs

```diff
@@ -52,62 +49,139 @@ public List<ExperimentalProteoform> identify_connected_experimentals()
 
                 double mass_tolerance = this.modified_mass / 1000000 * (double)Lollipop.mass_tolerance;
                 int sign = Math.Sign(e.modified_mass - modified_mass);
-                double deltaM = Math.Sign(r.peak_center_deltaM) < 0 ? r.peak_center_deltaM : sign * r.peak_center_deltaM; // give EE relations the correct sign, but don't switch negative ET relation deltaM's
-                ModificationWithMass best_addition = null;
+                double deltaM = Math.Sign(r.peak.peak_deltaM_average) < 0 ? r.peak.peak_deltaM_average : sign * r.peak.peak_deltaM_average; // give EE relations the correct sign, but don't switch negative ET relation deltaM's
+                TheoreticalProteoform theoretical_base = this as TheoreticalProteoform != null ?
+                    this as TheoreticalProteoform : //Theoretical starting point
+                    (linked_proteoform_references.First.Value as TheoreticalProteoform != null ?
+                        linked_proteoform_references.First.Value as TheoreticalProteoform : //Experimental with theoretical reference
+                        null); //Experimental without theoretical reference
```

Comment:
OK so I'm understanding it right - if a T is connected to 2 E's, it will trace a path through both of them and these will be separate linked_proteoform_references? And if an E is connected to two E's at some point, there could be two paths formed here? And any path goes until it reaches a dead end (can't identify a new one) or the end (no more E's to identify)